org.abapgit.adt.ui/src/org/abapgit/adt/ui/internal/wizards/AbapGitWizardPageRepositoryAndCredentials.java

```diff
@@ -319,10 +319,13 @@ public void run(IProgressMonitor monitor) throws InvocationTargetException, Inte
 
 			//if credentials are provided, ask user if the credentials should be stored in the secure storage
 			if (this.cloneData.user != null && this.cloneData.pass != null) {
+				if (this.repositoryCredentials == null
+						|| (this.repositoryCredentials != null && (!this.cloneData.pass.equals(this.repositoryCredentials.getPassword())
+								|| !this.cloneData.user.equals(this.repositoryCredentials.getUser())))) {
```

Comment:
In case of `pull` action, if credentials are retrieved from secure storage and a new password/PAT is not provided, the dialog asking if the user wants to store the credentials in secure storage is not shown.